networkx/algorithms/centrality/incremental_closeness.py

```diff
@@ -0,0 +1,140 @@
+"""
+Incremental closeness centrality measure.
+"""
+
+import networkx as nx
+from networkx.exception import NetworkXError
+from networkx.utils.decorators import not_implemented_for
+
+__author__ = 'Michael Lauria <michael.david.lauria@gmail.com>'
+__all__ = ['incremental_closeness_centrality']
+
+
+@not_implemented_for('directed')
+def incremental_closeness_centrality(G,
+                                     edge,
+                                     prev_cc=None,
+                                     insertion=True,
+                                     normalized=True):
+    """
+    Compute closeness centrality for nodes using level-based work filtering
+    as described in Incremental Algorithms for Closeness Centrality by Sariyuce,
+    A.E. ; Kaya, K. ; Saule, E. ; Catalyiirek, U.V.
+
+    Closeness centrality [1]_ of a node `u` is the reciprocal of the
+    sum of the shortest path distances from `u` to all `n-1` other nodes.
+    Since the sum of distances depends on the number of nodes in the
+    graph, closeness is normalized by the sum of minimum possible
+    distances `n-1`.
+
+    .. math::
+
+        C(u) = \frac{n - 1}{\sum_{v=1}^{n-1} d(v, u)},
+
+    where `d(v, u)` is the shortest-path distance between `v` and `u`,
+    and `n` is the number of nodes in the graph.
+
+    Notice that higher values of closeness indicate higher centrality.
+
+    Parameters
+    ----------
+    G : graph
+      A NetworkX graph
+    edge: tuple
+      The modified edge (u, v) in the graph.
+    prev_cc: list
```

Comment:
I think prev_cc is a dict.

Reply:
True, thanks. I don't see any issues about using mypy; is that in the cards?